### ISSUE TITLE
Elmo on Squad2 Not Converging

### ISSUE THREAD

**Issue by liuzzi** [kevinduh/san_mrc#11]
Hi! firstoff, thanks for this awesome repo.

So I've been playing with the new addition of Elmo embeddings with squad 2 and it doesnt seem to ever learn much.  accuracy and f1s are all around 50% and loss barely goes down after a certain point.  Do you think this is a hyperparam issue? Any ideas? 

Thanks

**Comment by namisan** [comment]
I haven't test on v2 yet, but can you share the log to me? I'll take a look. thanks.

**Comment by liuzzi** [comment]
For config I used the same as in the repo, except lowered dropout_p=0.1 and just trained with --elmo_on.

training loss creeps down about .03 per epoch
learning curve looks like this:


_[Epoch 0 - dev EM: 49.600 F1: 49.653 (best EM: 49.600 F1: 49.653)]
[Epoch 0 - ACC: 49.9284]
[Epoch 1 - dev EM: 49.701 F1: 49.783 (best EM: 49.701 F1: 49.783)]
[Epoch 1 - ACC: 49.9284]
[Epoch 2 - dev EM: 49.844 F1: 49.856 (best EM: 49.844 F1: 49.856)]
[Epoch 2 - ACC: 49.9284]
[Epoch 3 - dev EM: 50.013 F1: 50.013 (best EM: 50.013 F1: 50.013)]
[Epoch 3 - ACC: 49.9284]
[Epoch 4 - dev EM: 50.021 F1: 50.027 (best EM: 50.021 F1: 50.027)]
[Epoch 4 - ACC: 49.9284]_

**Comment by liuzzi** [comment]
full log

[san_output.txt](https://github.com/kevinduh/san_mrc/files/2537380/san_output.txt)

**Comment by bobycv06fpm** [comment]
Thanks for the log.
I worked on Elmo recently applying it with CNTK
Could you share your repo so that I can check where its not working?

**Comment by liuzzi** [comment]
All i did was follow the squad2 training directions from the master repo here, and trained with --elmo_on

**Comment by saket-maheshwary** [comment]
Hi @liuzzi @namisan,

While training on SQUAD2 (without ELMO) does the following command worked for you without any changes ? I am facing errors regarding the dev_gold labels directory. 

python train.py --v2_on --dev_gold data\dev-v2.0.json


Thanks

**Comment by liuzzi** [comment]
I don't believe i had problems with the labels directory, although i vaguely remember having to correct some Elmo errors.  If i get the chance to test the clean master repo again i'll update you

**Comment by saket-maheshwary** [comment]
Sure. 
I cloned the repo a couple of days back and was able to reproduce the results on v1.1, both with and without Elmo but got stuck on v2.0.

**Comment by saket-maheshwary** [comment]
@liuzzi 

For me the model is converging with Elmo on Squad2, though EM  and F1 scores are better without Elmo embeddings on same set of hyperparameters.

**Comment by namisan** [comment]
I didn't see the converge issues if you use an appropriate learning rate. Thus, I'll close this one.
